compiler/ballerina-lang/src/main/java/io/ballerina/projects/environment/PackageResolver.java

```diff
@@ -29,6 +34,10 @@
  */
 public interface PackageResolver {
 
+    Map<ImportModuleRequest, ImportModuleResponse> resolvePackageNames(List<ImportModuleRequest> importModuleRequests);
```

Comment:
How about returning a list of ImportModuleResponse and include the request details in the response. We did that with `ResolutionResponse`. Do you see disadvantageous with that approach?

Reply:
+1 let me do the change